### ISSUE TITLE
api/clients is blocked by CORS policy

### ISSUE THREAD

**Issue by roysbailey** [ardalis/pluralsight-ddd-fundamentals#35]
Hi,

I have been watching your Pluralsight Course and thought I would run the code locally in docker.  The docker build seemed to work OK, as did the compose up.  

However, when I launch the front end, I get the following error: 
![image](https://user-images.githubusercontent.com/3840972/137618133-917adac4-a45c-4a1a-ab53-00a4bd20395b.png)

Maybe I am missing something, but as far as I can see, i have followed the instructions.

Thanks.

**Comment by Jamanya** [comment]
@roysbailey  this is a known issue with chrome, you will either need to add a cors header to the backend, or add the CORS plugin to chome

**Comment by ardalis** [comment]
This one keeps biting me - but not all the time - as well. @Jamanya do you have more info on why Chrome is the culprit here?

**Comment by ardalis** [comment]
Related:
https://stackoverflow.com/questions/66534759/chrome-cors-error-on-request-to-localhost-dev-server-from-remote-site

Possible solution:
https://web.dev/cors-rfc1918-feedback/#step-2:-sending-preflight-requests-with-a-special-header

**Comment by ardalis** [comment]
It's weird because I ran into this (again) today. Changed nothing. Restarted docker services and tried again (getting ready to try to apply some of the fixes noted above) and everything worked fine the second time.

**Comment by hugh-maaskant** [comment]
I ran into the same? problem with Firefox (103.1).
In Clinic Management I was able to change a Clients preferred name (from David to Dave) without problem, but when I tried to change doctor Who's name (to WhoIs), I got an exception thrown in Blazor (see console screenshot and the log indicated an exception on the Doctor constructor (see log screenshot).

The link to the explanation in the console is: [https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin).
Basically it says: "CORS header 'Access-Control-Allow-Origin' missing".

![console](https://user-images.githubusercontent.com/6114240/183079122-62af519d-3b31-4015-a3be-0421b824a52c.png)
![log](https://user-images.githubusercontent.com/6114240/183079145-9bc01b53-91ad-446c-a737-f45507f489f4.png)
.
